### PR TITLE
feat: add other policies in CEL expressions - Part 6

### PR DESCRIPTION
**Pull request by Chandan-DK**
## Related Issue(s)
Partially addresses https://github.com/kyverno/policies/issues/891

## Description
This PR includes the conversion of policies in the other folder to Kyverno CEL policies. Conversion of the policies is done in multiple PRs

**Policies converted in this PR:**

- [x] restrict-loadbalancer
- [x] restrict-networkpolicy-empty-podselector
- [x] restrict-node-affinity
- [x] restrict-sa-automount-sa-token
- [x] restrict-secret-role-verbs
- [x] restrict-secrets-by-name
- [x] restrict-service-port-range
- [x] restrict-storageclass
- [x] restrict-usergroup-fsgroup-id
- [x] restrict-wildcard-resources
- [x] restrict-wildcard-verbs

## Checklist

- [] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

**Comment by MariamFahmy98** [comment]
@Chandan-DK - Could you please resolve the conflicts?

File: other-cel/restrict-usergroup-fsgroup-id/restrict-usergroup-fsgroup-id.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  - name: validate-userid-groupid-fsgroup
+    match:
+      any:
+      - resources:
+          kinds:
+          - Pod
+    validate:
+      cel:
+        expressions:
+          - expression: "has(object.spec.securityContext.runAsUser) && object.spec.securityContext.runAsUser == 1000"
```
[the remark this answers is not in the document]
Comment (Chandan-DK) [reply other-cel/restrict-usergroup-fsgroup-id/restrict-usergroup-fsgroup-id.yaml:31]: Let's create a pod without the `securityContext` field set.
```
cdk:~$ k apply -f -<<EOF
apiVersion: v1
kind: Pod
metadata:
  name: nginx
spec:
  containers:
  - image: nginx
    name: nginx
EOF
pod/nginx created
```

Output of the pod's spec.
```
cdk:~$ k get pods nginx -o yaml | yq .spec
containers:
  - image: nginx
    imagePullPolicy: Always
    name: nginx
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
      - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
        name: kube-api-access-fxwph
        readOnly: true
dnsPolicy: ClusterFirst
enableServiceLinks: true
nodeName: kind-control-plane
preemptionPolicy: PreemptLowerPriority
priority: 0
restartPolicy: Always
schedulerName: default-scheduler
securityContext: {}
serviceAccount: default
serviceAccountName: default
terminationGracePeriodSeconds: 30
tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
volumes:
  - name: kube-api-access-fxwph
    projected:
      defaultMode: 420
      sources:
        - serviceAccountToken:
            expirationSeconds: 3607
            path: token
        - configMap:
            items:
              - key: ca.crt
                path: ca.crt
            name: kube-root-ca.crt
        - downwardAPI:
            items:
              - fieldRef:
                  apiVersion: v1
                  fieldPath: metadata.namespace
                path: namespace
```

We see that `securityContext: {}` will be set by default. So, we can avoid the extra `has(object.spec.securityContext)` check. I've added tests for this case.

**Comment by Chandan-DK** [comment]
While resolving the merge conflict caused due to PR #1025, I realized that a `Role` and `ClusterRole` can be created by either omitting or keeping `rules` empty (https://github.com/kyverno/policies/pull/1025/files#diff-d8ebec1f6fac7038fbd1ecdea4d1797a84639e2fc96377af0537e5bc074fa200R1-R24). When it is created in this manner in the cluster, it will be set to `rules: null` by default. I've modified the policies to account for this behaviour and I've added for tests it.

**Comment by MariamFahmy98** [comment]
> While resolving the merge conflict caused due to PR #1025, I realized that a `Role` and `ClusterRole` can be created by either omitting or keeping `rules` empty (https://github.com/kyverno/policies/pull/1025/files#diff-d8ebec1f6fac7038fbd1ecdea4d1797a84639e2fc96377af0537e5bc074fa200R1-R24). When it is created in this manner in the cluster, it will be set to `rules: null` by default. I've modified the policies to account for this behaviour and I've added for tests it.

Good catch.

**Review by MariamFahmy98 — APPROVED** [review]

**Pull request merged**